packages/core/admin/admin/src/utils/tests/uniqueAdminHash.test.js

```diff
@@ -0,0 +1,55 @@
+import { Crypto } from '@peculiar/webcrypto';
+import { TextEncoder } from 'util';
+import hashAdminUserEmail, { utils } from '../uniqueAdminHash';
+
+const testHashValue = '8544bf5b5389959462912699664f03ed664a4b6d24f03b13bdbc362efc147873';
+
+describe('Creating admin user email hash in admin', () => {
+  afterEach(() => {
+    Object.defineProperty(global.self, 'crypto', {
+      value: undefined,
+      configurable: true,
+    });
+    Object.defineProperty(global.self, 'TextEncoder', {
+      value: undefined,
+      configurable: true,
+    });
+  });
+
+  it('should return empty string if no payload provided', async () => {
+    const testHash = await hashAdminUserEmail();
+
+    expect(testHash).toBe('');
+  });
+
+  it('should return hash using crypto subtle', async () => {
+    Object.defineProperty(global.self, 'crypto', {
```

Comment:
I'm not sure I get the point of having added the @peculiar/webcrypto package in our case. You only use it here for our tests, meaning that what you're doing is testing the validity of that package which we are not actually using.

Instead, why not just mock - once in that case - the `global.window.crypto` object? You don't even have to truly mock the implementation, the only important thing here is that the correct function gets called and that it returns a value of the correct type (an `ArrayBuffer` for the `digest` function in our case).

Reply:
Removed it